### PR TITLE
Support basic auth

### PR DESCRIPTION
**Pull request by thangnc**
**CHANGES**
- Support params for basic auth in download action

### CODE REVIEW

**Comment by thangnc** [comment]
@MartinGonzalez can you help to take a look?

File: lib/fastlane/plugin/json/actions/download_json_action.rb [hunk 1]
```diff
@@ -9,13 +9,26 @@ class DownloadJsonAction < Action
...
 
         begin
-          response = Net::HTTP.get_response(uri)
-          JSON.parse(response.body, symbolize_names: true)
+          Net::HTTP.start(uri.host, uri.port,
+                          use_ssl: uri.scheme == 'https',
+                          verify_mode: OpenSSL::SSL::VERIFY_NONE) do |http|
+
+            request = Net::HTTP::Get.new(uri.request_uri)
+            if !username.nil? && !username.empty? && !password.nil? && !password.empty?
```
Comment (MartinGonzalez) [inline lib/fastlane/plugin/json/actions/download_json_action.rb:25]: here you can use `!username.to_s.empty? && !password.to_s.empty?` to make to null or empty check in one :)

**Review by MartinGonzalez — CHANGES_REQUESTED** [review]
Just to check, have you tested this with and without authentication, to see that everything is still working?

File: lib/fastlane/plugin/json/actions/download_json_action.rb (hunk 1 above)
Comment (thangnc) [reply lib/fastlane/plugin/json/actions/download_json_action.rb]: Yeap, i updated the code as you commented

Comment (thangnc) [reply lib/fastlane/plugin/json/actions/download_json_action.rb]: > Just to check, have you tested this with and without authentication, to see that everything is still working?

Yeap, i checked on my Nexus URL with basic auth and it works like charm. But i found no free server to support basic auth to write unit test. Gist only supports signing in via OAuth, then i ignore the unit test

Comment (MartinGonzalez) [reply lib/fastlane/plugin/json/actions/download_json_action.rb]: Amazing, tomorrow I will add the new version with this changes! Thanks again for this collab!

File: lib/fastlane/plugin/json/actions/download_json_action.rb [hunk 2]
```diff
@@ -9,13 +9,25 @@ class DownloadJsonAction < Action
...
 
         begin
-          response = Net::HTTP.get_response(uri)
-          JSON.parse(response.body, symbolize_names: true)
+          Net::HTTP.start(uri.host, uri.port,
+                          use_ssl: uri.scheme == 'https',
+                          verify_mode: OpenSSL::SSL::VERIFY_NONE) do |http|
+
+            request = Net::HTTP::Get.new(uri.request_uri)
+            if !username.to_s.empty? && !password.to_s.empty?
```
Comment (MartinGonzalez) [inline lib/fastlane/plugin/json/actions/download_json_action.rb:25]: Is it possible you forgot to add the code that adds the auth to the request?

**Review by MartinGonzalez — CHANGES_REQUESTED** [review]

File: lib/fastlane/plugin/json/actions/download_json_action.rb (hunk 2 above)
Comment (thangnc) [reply lib/fastlane/plugin/json/actions/download_json_action.rb:25]: > Is it possible you forgot to add the code that adds the auth to the request?

Yeah, my mistake when copying code from the old PR to the new one. I updated it. PTAL

**Pull request closed without merging**